File: src/types/csvRecord.ts
export interface CsvPreviewRow {
  row: number;
  amount: string;
  bank_name: string;
  branch_name: string;
  description: string;
  edi: string;
}

export interface CsvValidationError {
  row: number;
  field: string;
  message: string;
}

export interface CsvReadResult {
  preview: CsvPreviewRow[];
  total_count: number;
  errors: CsvValidationError[];
}

File: src-tauri/src/domain/csv_record.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct CsvRecord {
    pub amount: u64,
    pub bank_name: String,
    pub branch_name: String,
    pub description: String,
    pub edi: String,
}

#[derive(Debug, Clone, Serialize)]
pub struct CsvValidationError {
    pub row: usize,
    pub field: String,
    pub message: String,
}
